[Microsoft.Devices] Added IotHubs API version 2025-08-01-preview

davidemontanari opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [X] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [x] Other, please clarify:
    - Added `deviceRegistry` property bag in `IotHubs` RT
    - Added `policyResourceId` property in `IotHubs/Cetrificates` RT
    - Added new GEN2 SKU for `IotHubs` RT

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.
- [ ] A [release plan](https://aka.ms/azsdkdocs/release-plans) has been created. If not, please create one as it will help guide you through the REST API and SDK creation process. 

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.
- For guidance on SDK breaking change review, refer to https://aka.ms/ci-fix.

davidemontanari committed:
Copy files from preview/2023-06-30-preview

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

davidemontanari committed:
Update version to preview/2025-05-01-preview

Updated the API version from preview/2023-06-30-preview to preview/2025-05-01-preview.

davidemontanari committed:
Added tag for 2025-05-01-preview in readme file

davidemontanari committed:
Added tlsCompatibilityMode property

davidemontanari committed:
Added location header for 202 responses

davidemontanari committed:
Fixed API version in examples

v-jiaodi committed:
Merge branch 'main' into davidemontanari/devices-hub-tlscompatmode

davidemontanari committed:
- Marked definitions.RoutingCosmosDBSqlApiProperties.properties.endpointUri property as readonly

- Specified definitions.RoutingCosmosDBSqlApiProperties.properties.endpointUri property to be of URL format

davidemontanari committed:
Merge branch 'main' into davidemontanari/devices-hub-tlscompatmode

davidemontanari committed:
Merge branch 'main' into davidemontanari/devices-hub-tlscompatmode

davidemontanari committed:
Added new example

ramoka178 commented:
>           "type": "boolean"

ARM recommends enums over booleans for future proof APIs. 

 

Standard guidance is: replace boolean/switch properties with a more meaningful enum whenever possible. 

A boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated]. 

Note: do NOT define a 'boolean enum' with two values 'True and False'. This might be easier to 'extend' in terms of types, but semantically its cofusing, and no better than a boolean. 

---
Refers to: specification/iothub/resource-manager/Microsoft.Devices/preview/2025-05-01-preview/iothub.json:2377 in 34b100c. [](commit_id = 34b100cb03f6b1ce20a242521260082c8779d6ae, deletion_comment = False)

davidemontanari committed:
Merge branch 'main' into davidemontanari/devices-hub-tlscompatmode

davidemontanari committed:
Merge branch 'main' into davidemontanari/devices-hub-tlscompatmode

davidemontanari committed:
Merge branch 'main' into davidemontanari/devices-hub-tlscompatmode

davidemontanari committed:
Merge branch 'main' into davidemontanari/devices-hub-tlscompatmode

davidemontanari committed:
Merge branch 'main' into davidemontanari/devices-hub-tlscompatmode

davidemontanari committed:
Merge branch 'main' into davidemontanari/devices-hub-tlscompatmode

davidemontanari committed:
Moved API version folder, examples and readme to 2025-08-01-preview

davidemontanari committed:
Added Azure Device Registry properties

davidemontanari committed:
- Marked definitions.RoutingCosmosDBSqlApiProperties.properties.endpointUri property as readonly

- Specified definitions.RoutingCosmosDBSqlApiProperties.properties.endpointUri property to be of URL format

davidemontanari committed:
Added examples

davidemontanari committed:
Merge branch 'main' into davidemontanari/devices-hub-tlscompatmode

davidemontanari committed:
Reverted readOnly and format for service bus queue properties

davidemontanari committed:
Merge branch 'main' into davidemontanari/devices-hub-tlscompatmode

davidemontanari committed:
Renamed AdrProperties to DeviceRegistry

davidemontanari committed:
Fixed example

davidemontanari committed:
Moved P SKUs to GEN2

davidemontanari committed:
Test for uppercase tier

davidemontanari committed:
Reverted tier casing in examples

davidemontanari committed:
Merge branch 'main' into davidemontanari/devices-hub-tlscompatmode

davidemontanari committed:
Updated tier value for GEN2 SKU

davidemontanari committed:
Merge branch 'main' into davidemontanari/devices-hub-tlscompatmode

davidemontanari committed:
Removed tlsCompatibilityMode property

davidemontanari committed:
Merge branch 'davidemontanari/devices-hub-tlscompatmode' of https://github.com/Azure/azure-rest-api-specs into davidemontanari/devices-hub-tlscompatmode

davidemontanari committed:
Merge branch 'main' into davidemontanari/devices-hub-tlscompatmode

davidemontanari committed:
Merge branch 'main' into davidemontanari/devices-hub-tlscompatmode

davidemontanari committed:
Merge branch 'main' into davidemontanari/devices-hub-tlscompatmode
